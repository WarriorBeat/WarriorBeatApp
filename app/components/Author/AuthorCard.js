/**
 * AuthorCard.js
 * Author Card Component
 * components/Author
 */

import React from "react"
import { observer, inject, PropTypes as MobxTypes } from "mobx-react/native"
import { PropTypes as gqlTypes } from "graphql"
import { Row, Col } from "components/Layout"
import Image from "react-native-fast-image"
import { StyleSheet, TouchableHighlight } from "react-native"
import { Shape, Colors } from "components/styles"
import { Typo } from "components/Text"

const styles = StyleSheet.create({
  image: {
    ...Shape.rect("30%", "26%", 6),
    elevation: 7,
    shadowOffset: { width: 5, height: 5 },
    shadowColor: Colors.get("ios.gray"),
    shadowOpacity: 0.5,
    shadowRadius: 10,
  },
})

const AuthorCard = inject("uiStore")(
  observer(({ author, uiStore }) => {
    const {
      profileImage, title, name, id,
    } = author
    const authorName = name.replace(" ", "\n")
    return (
      <Row height="20%" width="100%" vPad="2%" vMargin="5%">
        <Col center="end">
          <TouchableHighlight
            underlayColor="#fff"
            onPress={() => uiStore.push("Author.Profile", id, { author })}
          >
            <Image
              source={{ uri: profileImage.url }}
              style={styles.image}
              resizeMode={Image.resizeMode.cover}
            />
          </TouchableHighlight>
        </Col>
        <Col hPad="4.5%">
          <Row hPad="4%" size={2} width="50%" center="center" justify="flex-start">
            <Typo.TitleBlack>{authorName}</Typo.TitleBlack>
          </Row>
          <Row hPad="4%" size={1}>
            <Typo.SubtitleBlack>{title}</Typo.SubtitleBlack>
          </Row>
        </Col>
      </Row>
    )
  }),
)

AuthorCard.wrappedComponent.propTypes = {
  uiStore: MobxTypes.observableObject.isRequired,
}

AuthorCard.propTypes = {
  author: gqlTypes.author.isRequired,
}

export default AuthorCard
