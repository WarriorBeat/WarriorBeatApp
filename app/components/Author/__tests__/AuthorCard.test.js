/**
 * __tests__/AuthorCard.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { render } from "react-native-testing-library"
import AuthorCard from "../AuthorCard"
import Providers, { Data } from "tests"

test("should render correctly", () => {
  const tree = render(
    <Providers>
      <AuthorCard author={Data.author()} />
    </Providers>,
  )
  expect(tree).toMatchSnapshot()
})
