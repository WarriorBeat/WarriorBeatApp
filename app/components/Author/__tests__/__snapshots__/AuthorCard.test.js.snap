// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  center={null}
  hMargin="0%"
  hPad="0%"
  height="20%"
  justify="center"
  size={null}
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 0,
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "height": 267,
      "justifyContent": "center",
      "marginHorizontal": 0,
      "marginVertical": 66.5,
      "paddingHorizontal": 0,
      "paddingVertical": 26.5,
      "width": 750,
    }
  }
  vMargin="5%"
  vPad="2%"
  width="100%"
>
  <View
    center="end"
    hMargin="0%"
    hPad="0%"
    height={null}
    justify="center"
    size={null}
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "height": undefined,
        "justifyContent": "center",
        "marginHorizontal": 0,
        "marginVertical": 0,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": undefined,
      }
    }
    vMargin="0%"
    vPad="0%"
    width={null}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 6,
          "elevation": 7,
          "shadowColor": "#8E8E93",
          "shadowOffset": Object {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 6,
              "height": 347,
              "width": 225,
            },
          ]
        }
      >
        <FastImageView
          resizeMode="cover"
          source={
            Object {
              "uri": "https://s3.amazonaws.com/warriorbeat-graphql-media-dev/media/a816ba91-1b34-45d8-b9ea-774e58c9583e_d52b98fd-09f2-4ede-a1af-062a06e46d43.jpeg",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    center={null}
    hMargin="0%"
    hPad="4.5%"
    height={null}
    justify="center"
    size={null}
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "height": undefined,
        "justifyContent": "center",
        "marginHorizontal": 0,
        "marginVertical": 0,
        "paddingHorizontal": 34,
        "paddingVertical": 0,
        "width": undefined,
      }
    }
    vMargin="0%"
    vPad="0%"
    width={null}
  >
    <View
      center="center"
      hMargin="0%"
      hPad="4%"
      height={null}
      justify="flex-start"
      size={2}
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "marginHorizontal": 0,
          "marginVertical": 0,
          "paddingHorizontal": 30,
          "paddingVertical": 0,
          "width": 375,
        }
      }
      vMargin="0%"
      vPad="0%"
      width="50%"
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 22,
            "fontStyle": "normal",
            "fontWeight": "900",
            "includeFontPadding": true,
            "letterSpacing": 0.34375,
            "lineHeight": 28,
            "textTransform": "uppercase",
          }
        }
      >
        Myles
Hudson
      </Text>
    </View>
    <View
      center={null}
      hMargin="0%"
      hPad="4%"
      height={null}
      justify="center"
      size={1}
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "center",
          "marginHorizontal": 0,
          "marginVertical": 0,
          "paddingHorizontal": 30,
          "paddingVertical": 0,
          "width": undefined,
        }
      }
      vMargin="0%"
      vPad="0%"
      width={null}
    >
      <Text
        style={
          Object {
            "color": "#007AFF",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "900",
            "includeFontPadding": true,
            "textTransform": undefined,
          }
        }
      >
        Lead Designer
      </Text>
    </View>
  </View>
</View>
`;
